cluster.yml

```
@@ -40,6 +40,18 @@
     - { role: kubespray-defaults }
     - { role: bootstrap-os, tags: bootstrap-os}
 
+- name: Gather facts
+  hosts: localhost
+  gather_facts: false
+  become: no
+  tasks:
+    - name: Gather facts from ALL hosts (regardless of limit or tags)
+      setup:
+      delegate_to: "{{ item }}"
+      delegate_facts: True
+      when: hostvars[item]['ansible_default_ipv4'] is not defined
+      with_items: "{{ groups['all'] }}"
```

You need to ignore errors, else we can't run the playbook when a host is down